java/src/main/native/src/ColumnViewJni.cu

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/iterator.cuh>
+#include <cudf/detail/valid_if.cuh>
+
+#include "ColumnViewJni.hpp"
+
+namespace cudf::jni {
+
+std::unique_ptr<cudf::column> new_column_with_validity(cudf::column_view const &exemplar,
+                                                       cudf::column_view const &validity_column) {
+  CUDF_EXPECTS(validity_column.type().id() == type_id::BOOL8,
+               "Validity column must be of type bool");
+  CUDF_EXPECTS(validity_column.size() == exemplar.size(),
+               "Validity column must be of same size as exemplar column");
+
+  auto validity_device_view = cudf::column_device_view::create(validity_column);
+  auto validity_begin = cudf::detail::make_optional_iterator<bool>(
+      *validity_device_view, cudf::nullate::DYNAMIC{validity_column.has_nulls()});
+  auto validity_end = validity_begin + validity_device_view->size();
+  auto [null_mask, null_count] =
+      cudf::detail::valid_if(validity_begin, validity_end, [] __device__(auto optional_bool) {
+        return optional_bool.value_or(false);
+      });
```

Comment:
I find it a bit odd to see a custom kernel for what essentially is either a cudaMemsetAsync if the original column was not nullable or a cudaMemcpyAsync if it was.  The null count can be obtained from the original column, so it seems this could be completely done in a .cpp file.

Reply:
> a cudaMemsetAsync if the original column was not nullable

Scratch that, we'd just want to omit the validity mask in the new column for that case, so this just boils down to memcpy if the column had nulls.

Reply:
I've tried clarifying what's going on [here](https://github.com/rapidsai/cudf/pull/9901#discussion_r769291988). I'm afraid the `BOOL8` can't be memcpy-ed into a validity buffer.